src/lib.rs

```diff
@@ -667,7 +790,7 @@ impl Client {
     }
 
     /// Switches to the frame specified at the index.
-    pub async fn enter_frame(mut self, index: Option<u16>) -> Result<Client, error::CmdError> {
+    pub async fn enter_frame(mut self, index: Option<u16>) -> Result<session::Client<C>, error::CmdError> {
```

Comment:
Can't this just be `Client<C>`? Given how often we refer to `Client` in this file, I think we should just continue importing it the way we used to.